crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -445,6 +438,21 @@ impl EventHandler {
                     return;
                 }
 
+                let fake_channel_payments = self.fake_channel_payments.clone();
+                let result = fake_channel_payments.lock();
+                let guard = result.unwrap();
+
+                let target_node_id = match guard.get(&requested_next_hop_scid) {
```

Comment:
🤔 not sure if this is 100% correct. What if the `requested_next_hop_scid` isn't a fake one, because the user wants to pay to an known node?

cc: @bonomat @luckysori

Reply:
Then we should not trigger this event. From the docs:


>Used to indicate that we've intercepted an HTLC forward. This event will only be generated if you've encoded an intercept scid in the receiver's invoice route hints using `ChannelManager::get_intercept_scid`

Reply:
i see, I though any HTLC will be intercepted by this event. 👍